src/JabbR-Core/Commands/CommandManager.cs

```diff
@@ -0,0 +1,214 @@
+            }
+
+            command.Execute(context, callerContext, args);
+
+            return true;
+        }
+
+        private void MatchCommand(string commandName, out ICommand command)
+        {
+            throw new NotImplementedException();
+        }
+
+
+        //public void MatchCommand(string commandName, out ICommand command)
+        //{
+        //    if (_commandCache == null)
+        //    {
+        //        var commands = from c in _commands.Value
+        //                       let commandAttribute = c.GetType()
+        //                                               .GetCustomAttributes(true)
+        //                                               .OfType<CommandAttribute>()
+        //                                               .FirstOrDefault()
+        //                       where commandAttribute != null
+        //                       select new
+        //                       {
+        //                           Name = commandAttribute.CommandName,
+        //                           Command = c
+        //                       };
+
+        //        _commandCache = commands.ToDictionary(c => c.Name,
+        //                                              c => c.Command,
+        //                                              StringComparer.OrdinalIgnoreCase);
+        //    }
+
+        //    IList<string> candidates = null;
+
+        //    var exactMatches = _commandCache.Keys.Where(comm => comm.Equals(commandName, StringComparison.OrdinalIgnoreCase))
+        //                                         .ToList();
+
+        //    if (exactMatches.Count == 1)
+        //    {
+        //        candidates = exactMatches;
+        //    }
+        //    else
+        //    {
+        //        candidates = _commandCache.Keys.Where(comm => comm.StartsWith(commandName, StringComparison.OrdinalIgnoreCase))
+        //                                       .ToList();
+        //    }
+
+        //    switch (candidates.Count)
+        //    {
+        //        case 1:
+        //            _commandCache.TryGetValue(candidates[0], out command);
+        //            commandName = candidates[0];
+        //            break;
+        //        case 0:
+        //            throw new CommandNotFoundException();
+        //        default:
+        //            throw new CommandAmbiguityException(candidates);
+        //    }
+        //}
+
+        private static IList<ICommand> GetCommands()
+        {
+            // Use MEF to locate the content providers in this assembly
+            //var catalog = new AssemblyCatalog(typeof(CommandManager).Assembly);
+            //var compositionContainer = new CompositionContainer(catalog);
+            //return compositionContainer.GetExportedValues<ICommand>().ToList();
+            return new List<ICommand>() { new JoinCommand() };
+        }
```

Comment:
@adamtuliper This was the workaround we used. We Manually populated JoinCommand() in the List<ICommand>.
Lines 143 and 199 use `using System.ComponentModel.Composition.Hosting;` which is not supported by .NET Core. 

Currently trying to find workaround using System.Reflection